Task1/Container.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Metadata;
+using Task1.DoNotChange;
+
+namespace Task1
+{
+    public class Container
+    {
+        private Dictionary<Type, Type> _serviceDescriptors = new Dictionary<Type, Type>();
+
+        public void AddAssembly(Assembly assembly)
+        {
+            var types = assembly.GetTypes();
+            foreach (var type in types)
+            {
+                var exportAttribute = type.GetCustomAttribute<ExportAttribute>();
+                var importProperties = type.GetProperties().Where(prop => prop.GetCustomAttribute<ImportAttribute>() != null);
+                if (type.GetCustomAttribute<ImportConstructorAttribute>()!=null)
+                {
+                    _serviceDescriptors.Add(type, type);
+                }
+                else if (exportAttribute != null)
+                {
+                    if (exportAttribute.Contract == null)
+                    {
+                        _serviceDescriptors.Add(type, type);
+                    }
+                    else
+                    {
+                        _serviceDescriptors.Add(exportAttribute.Contract, type);
+                    }
+                }
+                else if(importProperties.Any())
+                {
+                    _serviceDescriptors.Add(type, type);
+                }
+            }
+        }
+
+        public void AddType(Type type)
+        {
+            _serviceDescriptors.Add(type, type);
+        }
+
+        public void AddType(Type type, Type baseType)
+        {
+            _serviceDescriptors.Add(baseType, type);
+        }
+
+        public T Get<T>()
+        {
+            var implementationType =
+                _serviceDescriptors.LastOrDefault(serviceDescriptor => serviceDescriptor.Key == typeof(T) || serviceDescriptor.Value == typeof(T));
+            var constructorParameters = implementationType.Value.GetConstructors()[0].GetParameters();
+            if (constructorParameters.Length > 0)
+            {
+                var exportAttribute = implementationType.Value.GetCustomAttribute<ExportAttribute>();
+                var importConstructorAttribute =
+                    implementationType.Key.GetCustomAttribute<ImportConstructorAttribute>();
```

Comment:
Don't use value-pair everywhere, you should have T as key type and your Value as potential implementation type.